File: _java/symulacja_organizmow/src/Symulacja/Rosliny/WilczeJagody.java
package Symulacja.Rosliny;
import Symulacja.*;

import java.awt.*;
import java.util.Set;

public class WilczeJagody extends Roslina {
   public WilczeJagody(Swiat swiat){
        super(swiat);
       SetSila();
    };
    public char Rysowanie(){
     return 'w';
    };
    public Organizm Urodz(){
     return new WilczeJagody(this.swiat);
    };
    public int GetPrawdopodobienstwo(){
     return 5;
    };
    public boolean MozeBycPartnerem(Organizm organizm){
      if(organizm instanceof WilczeJagody)
       return true;
      else
       return false;
    };
    public void Kolizja(Organizm organizm){
      swiat.zdarzenia.DodajString("Organizm: " + organizm.Rysowanie() + " na pozycji " + organizm.GetPolozenie() + " <br/>zjada Wilcze Jagody i umiera");
      swiat.ZabijOrganizm(organizm);
    }
    protected void SetSila(){
      this.sila=99;
    }
    @Override
    public Color GetColor(){
        return new Color(255,0,255);
    };
}

File: _c++/symulacja_organizmow/Organizmy/Rosliny/WilczeJagody.cpp
//
//  WilczeJagody.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "WilczeJagody.hpp"

WilczeJagody::WilczeJagody(Swiat &swiat):Roslina(swiat){

}
void WilczeJagody::Kolizja(Organizm& organizm){
    std::cout << "Organizm: " << organizm.Rysowanie() << " na pozycji " << organizm.GetPolozenie() << " zjada Wilcze Jagody i umiera" << std::endl;
    swiat.ZabijOrganizm(organizm);
}
char WilczeJagody::Rysowanie() const{
    return 'w';
}
void WilczeJagody::SetSila(){
    this->sila=99;
}
Organizm* WilczeJagody::Urodz(){
    return new WilczeJagody(this->swiat);
}
int WilczeJagody::GetPrawdopodobienstwo(){
    return 5;
}
bool WilczeJagody::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<WilczeJagody*>(&organizm))
       return true;
    else
       return false;
}
